image_geometry/src/pinhole_camera_model.cpp

```diff
@@ -56,11 +72,11 @@ bool update(const T& new_val, T& my_val)
   return true;
 }
 
-// For boost::array, std::vector
+// For std::array, std::vector
 template<typename MatT>
 bool updateMat(const MatT& new_mat, MatT& my_mat, cv::Mat_<double>& cv_mat, int rows, int cols)
 {
-  if ((my_mat == new_mat) && (my_mat.size() == cv_mat.rows*cv_mat.cols))
+  if ((my_mat == new_mat) && (my_mat.size() == (unsigned)(cv_mat.rows*cv_mat.cols)))
```

Comment:
Looking at `cv::Mat`, I think `static_cast<unsigned int>` would work, but I'm not 100% sure.